pinc/Round.inc

```diff
@@ -546,7 +546,7 @@ function round_project_listing_presort($round)
             ) DESC
         ";
     } else {
-        assert(false);
+        assert(false); /** @phpstan-ignore-line */
```

Comment:
I think we should rethink the original approach to using asserts like this that are intended to detect developer errors vs user errors. For cases like this where it's a param passed in with a known set, throwing `UnexpectedValueException` might be a better approach.

Reply:
OK. I've sprinkled a bunch of `UnexpectedValueException`s and `LogicException`s.